cmd/filegen/main.go

```diff
@@ -394,50 +391,13 @@ func generateFiles(ctx *cli.Context) error {
 		if err != nil {
 			return err
 		}
-	}
-
-	genesisBuff, err := json.MarshalIndent(genesis, "", "  ")
-	if err != nil {
-		return err
-	}
-
-	_, err = genesisFile.Write(genesisBuff)
-	if err != nil {
-		return err
-	}
-
-	// TODO: A factory which returns the suite according to consensus type should be created in elrond-go project
-	// Ex: crypto.NewSuite(consensusType) crypto.Suite
-	switch consensusType {
-	case "bls":
-		suite = kyber.NewSuitePairingBn256()
-	case "bn":
-		suite = kyber.NewBlakeSHA256Ed25519()
-	default:
-		suite = nil
-	}
-
-	initialNodesKeyGenerator = signing.NewKeyGenerator(suite)
-	numObservers := numOfShards*numOfObserversPerShard + numOfMetachainObservers
-	for i := 0; i < totalNumOfNodes+numObservers; i++ {
-		pkHex, skHex, err = getIdentifierAndPrivateKey(initialNodesKeyGenerator)
-		if err != nil {
-			return err
-		}
 
-		addrPkHex, _, err = getIdentifierAndPrivateKey(balancesKeyGenerator)
+		pkHexForNode, skHexForNode, err := getIdentifierAndPrivateKey(getNodesKeyGen(consensusType))
```

Comment:
You should extract in variable the return of getNodesKeyGen(consensusType), as this method could return nil, and in this case you would get panic in first line of getIdentifierAndPrivateKey method

Reply:
done